src/lib/lore.sh

```diff
@@ -312,67 +320,75 @@ function processing_new_patches()
   local xpath_query
   local count=0
   local index=0
+  local starting_index=0
   local title
-  local url_filter='?q=d:2.day.ago..+AND+NOT+s:Re&x=A'
-  local default_url="${LORE_URL}${target_mailing_list}/${url_filter}"
+  local query_filter
+  local query_url
+  local days="$TIMEFRAME_IN_DAYS"
+  local last_patch_timestamp=''
   local shared_dir_for_parallelism
   local list_patches_file_name="list-patches-${target_mailing_list}.xml"
 
-  download "$default_url" "$list_patches_file_name" "$CACHE_LORE_DIR" "$flag" || return "$?"
-
-  raw_list_path="${CACHE_LORE_DIR}/${list_patches_file_name}"
-
-  xpath_query="${NAME_EXP}|${EMAIL_EXP}|${TITLE_EXP}|${LINK_EXP}"
-
-  pre_processed=$(< "$raw_list_path")
-  pre_processed=$(printf '%s' "$pre_processed" | xpath -q -e "$xpath_query")
-
-  # Converting to:
-  #  Author, email, version, total patches, patch title, link
-  shared_dir_for_parallelism=$(mktemp -d)
-  while IFS= read -r line; do
-    if [[ "$line" =~ href ]]; then
-      message_id_link=$(str_get_value_under_double_quotes "$line")
-
-      if is_introduction_patch "$message_id_link"; then
-        # Process each patch in parallel
-        thread_for_process_patch "$index" "$shared_dir_for_parallelism" \
-          "$processed_line" "$message_id_link" "$title" &
-        ((index++))
+  while [[ "$index" -lt "$NUMBER_OF_PATCHSETS_FETCHED" ]]; do
+    query_filter="?q=rt:${days}.day.ago..${last_patch_timestamp}+AND+NOT+s:Re&x=A"
+    query_url="${LORE_URL}/${target_mailing_list}/${query_filter}"
```

Comment:
I was wondering if we should create a new function to build the URL and ensure it is valid inside this new function (e.g., `compose_url_with_verification()`).